File: Difficulty: Easy/Reverse a linked list/reverse-a-linked-list.py
"""
class Node:
    def __init__(self, val):
        self.data = val
        self.next = None
"""

class Solution:
    def reverseList(self, head):
        # Code here
        prev = next = None 
        current = head
        
        while (current) :
            next = current.next
            current.next = prev
            prev = current 
            current = next
            
        head = prev
        return head
        

File: Difficulty: Easy/Reverse a linked list/reverse-a-linked-list.cpp
/*
class Node {
 public:
    int data ;
    Node *next ;

    Node(int x) {
        data = x ;
        next = nullptr ;
    }
};
*/

class Solution {
  public:
    Node* reverseList(Node* head) {
        // code here
        Node *prev = nullptr;
        Node *current = head;
        Node *next = head;
        
        while(current != nullptr) {
            next = current->next;
            current->next = prev;
            prev = current;
            current = next;
        }
        head = prev;
        
        return head;
    }
};